src/test/java/com/ocient/jdbc/JDBCBenchmarks.java

```diff
@@ -0,0 +1,202 @@
+                .build();
+        Runner r = new Runner(opt);
+        r.run();
+    }
+
+    private static String m_url = "";
+
+    private static AtomicInteger teardownCounter = new AtomicInteger();
+
+    @Param({"blocking", "nonblocking"})
+    private static String m_transport;
+    private static Properties m_properties = new Properties();;
+
+    private static Connection m_conn;
+    private static Statement m_stmt;
+
+
+    private static final Logger LOGGER = Logger.getLogger("com.ocient.test"); // Test logger.
+
+    @Setup(org.openjdk.jmh.annotations.Level.Trial)
+    public void beforeClass() {
+        m_url = "jdbc:ocient://localhost:4050/test";
+        m_properties.setProperty(ConnectionProperty.USER.key(), "admin@system");
+        m_properties.setProperty(ConnectionProperty.PASSWORD.key(), "admin");
+        m_properties.setProperty("force", "true");
+        m_properties.setProperty("transport", m_transport.equals("blocking") ? "java.net" : "netty");
+    }
+
+    // Runs before each test.
+    @Setup(org.openjdk.jmh.annotations.Level.Iteration)
+    public void beforeTest() {
+        teardownCounter.getAndIncrement();
+        try {
+            m_conn = DriverManager.getConnection(m_url, m_properties);
+        } catch (final SQLException e){
+            LOGGER.log(Level.WARNING, "Failed to get a connection from driver manager on setup");
+            m_conn = null;
+            fail(e.toString());
+        }
+        try {
+            m_stmt = m_conn.createStatement();
+        } catch (final SQLException e){
+            LOGGER.log(Level.WARNING, "Failed to create statement on setup");
+            m_stmt = null;
+            m_conn = null;
+            fail(e.toString());
+        }
+    }
+
+    // Runs after each test.
+    @TearDown(org.openjdk.jmh.annotations.Level.Iteration)
+    public void afterTest() {
+        teardownCounter.getAndDecrement();
+    }
+
+    @TearDown(org.openjdk.jmh.annotations.Level.Trial)
+    public void afterClass() {
+        while (teardownCounter.get() > 0) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                fail(e.toString());
+            }
+        }
+
+        if (m_stmt == null || m_conn == null) {
+            return;
+        }
+
+        try {
+            m_stmt.close();
+        } catch (SQLException e){
+            LOGGER.log(Level.WARNING, "Failure to close statement on test teardown");
+        }
+        m_stmt = null;
+
+        try {
+            m_conn.close();
+        } catch (SQLException e){
+            LOGGER.log(Level.WARNING, "Failure to close statement on test teardown");
+        }
+        m_conn = null;
+    }
+
+    public ResultSet executeAndRetrieveUtility(String sqlStatement) {
+        ResultSet rs = null;
+        try {
+            m_stmt = m_conn.createStatement();
+            m_stmt.execute(sqlStatement);
+            rs = m_stmt.getResultSet();
```

Comment:
We also need to consume all elements in the result set.
```suggestion
            rs = m_stmt.getResultSet();
            while (rs.next()) {
              // consume all rows
            }
```